### PR TITLE
[1.16] LivingSpawnEvent.InitialSpawn

### PR DESCRIPTION
**Pull request by seymourimadeit**
This PR adds LivingSpawnEvent.InitialSpawn.

This event would get fired every time an entity is first spawned (or when MobEntity#OnInitialSpawn is called).

Included is a test mod.

### CODE REVIEW

File: src/test/java/net/minecraftforge/debug/entity/InitialSpawnTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    @SubscribeEvent
+    public static void doStuff(LivingSpawnEvent.InitialSpawn event) 
+    {
+      if (event.getEntity() instanceof CowEntity && ENABLE)
+      {
+    	  CowEntity cow = (CowEntity)event.getEntity();
+    	  SheepEntity sheep = EntityType.SHEEP.create(cow.world);
+    	  sheep.copyLocationAndAnglesFrom(cow);
+    	  cow.startRiding(sheep);
+    	  cow.world.addEntity(sheep);
```
Comment (pau101) [inline src/test/java/net/minecraftforge/debug/entity/InitialSpawnTest.java]: The `cow.world` should instead be `event.getWorld()` as the entity this event may fire for could be created from structure generation where the world is `WorldGenRegion`.

File: patches/minecraft/net/minecraft/entity/MobEntity.java.patch [hunk 2]
```diff
@@ -50,3 +50,12 @@
           }
        } else {
           return EquipmentSlotType.HEAD;
+@@ -972,7 +983,7 @@
+       } else {
+          this.func_184641_n(false);
+       }
+-
```
Comment (ichttt) [inline patches/minecraft/net/minecraft/entity/MobEntity.java.patch]: Please keep the newline, to reduce patches

File: src/test/java/net/minecraftforge/debug/entity/InitialSpawnTest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    {
+		
+    }
+    
+    @SubscribeEvent
+    public static void doStuff(LivingSpawnEvent.InitialSpawn event) 
+    {
+      if (event.getEntity() instanceof CowEntity && ENABLE)
+      {
+    	  CowEntity cow = (CowEntity)event.getEntity();
```
Comment (ichttt) [inline src/test/java/net/minecraftforge/debug/entity/InitialSpawnTest.java]: please use 4 spaces per indentation level in forge code

File: src/test/java/net/minecraftforge/debug/entity/InitialSpawnTest.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import net.minecraftforge.event.entity.living.LivingSpawnEvent;
+import net.minecraftforge.event.entity.living.LivingSpawnEvent.InitialSpawn;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+
+@Mod("initial_spawn_test")
+@Mod.EventBusSubscriber()
+public class InitialSpawnTest 
+{
+    public static final Boolean ENABLE = false;
```
Comment (ichttt) [inline src/test/java/net/minecraftforge/debug/entity/InitialSpawnTest.java]: this can be a primitive `boolean`

**Review by ichttt — CHANGES_REQUESTED** [review]
Also note pau101's comment in addition to this feedback

File: src/test/java/net/minecraftforge/debug/entity/InitialSpawnTest.java (hunk 3 above)
Comment (seymourimadeit) [reply src/test/java/net/minecraftforge/debug/entity/InitialSpawnTest.java]: fixed

**Comment by TheCurle** [comment]
This needs a rebase, a review and some more feedback.
A more descriptive title would help, too.

**Pull request closed without merging**